app/scripts/views/confirm_account_unlock.js

```diff
@@ -0,0 +1,153 @@
+
+    initialize: function () {
+      var data = this.ephemeralData();
+      var accountData;
+
+      if (data) {
+        if (data.lockoutSource) {
+          this._lockoutSource = data.lockoutSource;
+        }
+
+        if (data.account) {
+          accountData = data.account;
+        }
+      }
+
+      this._account = this.user.initAccount(accountData);
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    context: function () {
+      return {
+        email: this.getAccount().get('email')
+      };
+    },
+
+    events: {
+      // validateAndSubmit is used to prevent multiple concurrent submissions.
+      'click #resend': BaseView.preventDefaultThen('validateAndSubmit')
+
+    },
+
+    beforeRender: function () {
+      // browsing directly to the page should not be allowed.
+      var self = this;
+      return p().then(function () {
+        if (self.getAccount().isEmpty()) {
+          self.navigate('signup');
+          return false;
+        }
+      });
+    },
+
+    afterVisible: function () {
+      var graphic = this.$el.find('.graphic');
+      graphic.addClass('pulse');
+
+      var self = this;
+      return self.broker.persist()
+        .then(function () {
+          return self._waitForConfirmation();
+        })
+        .then(function (updatedSessionData) {
+          self.getAccount().set(updatedSessionData);
+          self.logScreenEvent('verification.success');
+
+          // the continuation path depends on the action that triggerred
+          // the account lockout notice. The only time the broker should
+          // be notified is if the user was trying to sign in.
+          if (/signin/.test(self._lockoutSource)) {
+            return self.broker.afterSignIn(self.getAccount())
+              .then(function (result) {
+                if (! (result && result.halt)) {
+                  self.navigate('account_unlock_complete');
+                }
+              });
+          }
+
+          // return non-signin users back to where they came from.
+          self.ephemeralMessages.set(
+              'success', t('Account unlocked, please try again'));
+          self.back();
+        })
+        .fail(self.displayError.bind(self));
+    },
+
+    _waitForConfirmation: function () {
+      var self = this;
+      var account = self.getAccount();
+      var email = account.get('email');
+      var password = account.get('password');
+      // try to sign the user in using the email/password that caused the
+      // account to be locked. If the user has verified their email address,
+      // the sign in will successfully complete. If they have not verified
+      // their address, the sign in call will fail with the ACCOUNT_LOCKED
+      // error, and we poll again.
+
+      return self.fxaClient.signIn(email, password, self.relier)
+        .then(null, function (err) {
+          if (AuthErrors.is(err, 'ACCOUNT_LOCKED')) {
```

Comment:
Whether the account is locked is checked before the password is correct. If the error is `INCORRECT_PASSWORD`, we know the account is unlocked, but the user typed in their password incorrectly - boot the user back to `/signin` to let them sign in. Add a functional test for this too.